File: Backend/pki-system/src/main/java/com/ftnteam11_2025/pki/pki_system/certificates/dto/RevocationReason.java
package com.ftnteam11_2025.pki.pki_system.certificates.dto;

import lombok.Getter;

@Getter
public enum RevocationReason {
    UNSPECIFIED(0),
    KEY_COMPROMISE(1),
    CA_COMPROMISE(2),
    AFFILIATION_CHANGED(3),
    SUPERSEDED(4),
    CESSATION_OF_OPERATION(5),
    CERTIFICATE_HOLD(6),
    REMOVE_FROM_CRL(8),
    PRIVILEGE_WITHDRAWN(9),
    AA_COMPROMISE(10);

    private final int code;

    RevocationReason(int code) {
        this.code = code;
    }

}

File: Frontend/src/app/certificates/model/revocation-reason.ts
export enum RevocationReason {
  Unspecified = 'UNSPECIFIED',
  KeyCompromise = 'KEY_COMPROMISE',
  CaCompromise = 'CA_COMPROMISE',
  AffiliationChanged = 'AFFILIATION_CHANGED',
  Superseded = 'SUPERSEDED',
  CessationOfOperation = 'CESSATION_OF_OPERATION',
  CertificateHold = 'CERTIFICATE_HOLD',
  RemoveFromCrl = 'REMOVE_FROM_CRL',
  PrivilegeWithdrawn = 'PRIVILEGE_WITHDRAWN',
  AaCompromise = 'AA_COMPROMISE'
}
